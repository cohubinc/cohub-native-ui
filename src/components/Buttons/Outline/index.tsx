import React from "react";
import { Color, IColor } from "@cohubinc/cohub-utils";

import TButtonProps from "src/definitions/interfaces/IButtonProps";
import BaseButton from "../Base";

const defaultHeight = 40;

export interface IOutlineButtonProps extends TButtonProps {
  bordered?: boolean;
  dark?: boolean;
  textColor?: IColor;
  outlineColor?: IColor;
}

const defaultProps: Partial<IOutlineButtonProps> = {
  bordered: true,
  dark: false,
  backgroundColor: Color.trueWhite,
  color: Color.iconGrey,
  enableHaptics: false
};

export default function Outline(props: IOutlineButtonProps) {
  const mergedProps = { ...defaultProps, ...props };

  const {
    style,
    bordered,
    dark,
    outlineColor,
    color,
    textColor,
    ...rest
  } = mergedProps;

  const backgroundColor = dark ? Color.black : mergedProps.backgroundColor;

  return (
    <BaseButton
      {...rest}
      style={[
        {
          backgroundColor,
          height: defaultHeight
        },
        style
      ]}
      labelStyle={{
        fontSize: 12,
        lineHeight: 14,
        color: (textColor || color) as any
      }}
      loaderColor={color}
      borderColor={(outlineColor || color) as any}
    />
  );
}
