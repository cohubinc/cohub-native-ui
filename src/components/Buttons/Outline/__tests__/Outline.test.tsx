import React from "react";
import { mount } from "enzyme";

import Outline from "../index";

describe("Outline Button", () => {
  it("renders without crashing", () => {
    const wrapper = mount(<Outline>push me</Outline>);
    expect(wrapper).toMatchSnapshot();
  });
});
