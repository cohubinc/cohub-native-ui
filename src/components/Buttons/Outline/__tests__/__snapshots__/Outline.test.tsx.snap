// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Outline Button renders without crashing 1`] = `
<Outline>
  <OutlineButtonBase>
    <Base
      backgroundColor="hsl(0, 0%, 100%)"
      raised={false}
      style={
        Object {
          "border": ".75px solid hsl(0, 3%, 7%)",
          "color": "hsl(0, 3%, 7%)",
        }
      }
    >
      <button
        className="CohubButton "
        style={
          Object {
            "backgroundColor": "hsl(0, 0%, 100%)",
            "border": ".75px solid hsl(0, 3%, 7%)",
            "boxShadow": undefined,
            "color": "hsl(0, 3%, 7%)",
          }
        }
      >
        <div
          className="button-text relative"
        >
          <Typography
            color="hsl(0, 3%, 7%)"
            secondary={true}
            style={
              Object {
                "opacity": 1,
                "transition": "opacity 150ms ease-in",
              }
            }
            uppercase={true}
          >
            <Component
              color="hsl(0, 3%, 7%)"
              secondary={true}
              style={
                Object {
                  "opacity": 1,
                  "transition": "opacity 150ms ease-in",
                }
              }
              uppercase={true}
            >
              <span
                className=""
                data-qa="text"
                style={
                  Object {
                    "color": "hsl(0, 3%, 7%)",
                    "fontFamily": "Akkurat-Mono",
                    "fontSize": "14px",
                    "fontWeight": 400,
                    "opacity": 1,
                    "textAlign": "initial",
                    "textTransform": "uppercase",
                    "transition": "opacity 150ms ease-in",
                  }
                }
              >
                <span>
                  push me
                </span>
              </span>
            </Component>
          </Typography>
        </div>
      </button>
    </Base>
  </OutlineButtonBase>
</Outline>
`;
